zad2/warriors.rb

```diff
@@ -1,12 +1,78 @@
-class Warrior
+class Character
+  
+  attr_accessor :name
+  attr_accessor :level
+
   def initialize(name:, level:)
+    self.name=name
+    self.level=level
   end
 
   def strength
+    [*1..12].sample + self.level
+  end
+
+  def card
+    puts"Name: #{name}, Level: #{level}"
   end
 end
 
+class Warrior < Character
+end
+class Monster < Character
+end
+
 class BattleArena
+  
+  attr_accessor :first_character
+  attr_accessor :second_character
+
   def initialize(first_character, second_character)
+    self.first_character=first_character
+    self.second_character=second_character
+  end
+
+  def fight(winner,loser)
```

Comment:
podobny problem jak poniżej - jest to metoda bardziej zajmująca się podnoszeniem poziomu postaci niż samą walką.

tip: metodę można uprościć przez użycie tzn `guard clause`
https://www.thechrisoshow.com/2009/02/16/using-guard-clauses-in-your-ruby-code/